tests/unit/externalRequestsTest.js

```diff
@@ -0,0 +1,154 @@
+'use strict'
+
+const rewire = require('rewire')
+const tap = require('tap')
+
+const externalRequests = rewire('../../src/middleware/externalRequests')
+
+tap.test('External Requests Middleware', {autoend: true}, t => {
+  t.test('prepareLookupRequests', {autoend: true}, t => {
+    t.test('should throw an error if any promise in the array fails', t => {
+      t.plan(2)
+
+      const performRequestsStub = externalRequests.__set__(
+        'performRequests',
+        () => [Promise.resolve('Success'), Promise.reject('Fail')]
+      )
+
+      const ctx = {
+        state: {
+          uuid: 'randomUidForRequest',
+          metaData: {
+            name: 'Testing endpoint',
+            requests: {
+              lookup: [
+                {
+                  // first lookup request config - responds with success
+                },
+                {
+                  // second lookup request config - responds with fail
+                }
+              ]
+            }
+          }
+        }
+      }
+
+      const prepareLookupRequests = externalRequests.__get__(
+        'prepareLookupRequests'
+      )
+
+      prepareLookupRequests(ctx)
+        .then(() => {
+          t.error('Should not reach here')
+        })
+        .catch(err => {
+          performRequestsStub()
+          t.type(err, Error)
+          t.equal(err.message, 'Rejected Promise: Fail')
+        })
+    })
+
+    t.test(
+      'should add response data to the ctx when all promises resolve',
+      t => {
+        t.plan(1)
+
+        const performRequestsStub = externalRequests.__set__(
+          'performRequests',
+          () => [
+            Promise.resolve({test1: 'testA'}),
+            Promise.resolve({test2: 'testB'})
+          ]
+        )
+
+        const ctx = {
+          state: {
+            uuid: 'randomUidForRequest',
+            metaData: {
+              name: 'Testing endpoint',
+              requests: {
+                lookup: [
+                  {
+                    // first lookup request config - responds with success
+                  },
+                  {
+                    // second lookup request config - responds with success
+                  }
+                ]
+              }
+            }
+          }
+        }
+
+        const prepareLookupRequests = externalRequests.__get__(
+          'prepareLookupRequests'
+        )
+
+        prepareLookupRequests(ctx)
+          .then(() => {
+            t.same({test1: 'testA', test2: 'testB'}, ctx.lookupRequests)
+            performRequestsStub()
+          })
+          .catch(() => {
+            t.error('Should not reach here')
```

Comment:
Im not sure the `t.error` does what you think it does. This function requires an object as the first argument which if this object is an error it fails the assertion.

The approach for this is also not correct due to the `t.plan(1)`. If the logic reached the `catch` and the assertion was valid it would pass the test as one planned test was executed succesfully

Reply:
Done in https://github.com/jembi/openhim-mediator-mapping/pull/37/commits/55bd803d2b2d2f076559cb6393703691621667c7